### PR TITLE
fix sequence

### PR DESCRIPTION
**Pull request by marchsu**
### What changed? Why?
- fix the [parameters sequence ](https://github.com/coinbase/coinbase-sdk-nodejs/blob/master/src/client/api.ts#L4374)to align with what `api.ts` has 
`networkId, assetId, addressId`

- modify `toString()` of `StakingBalance` to be more readable

#### Result
```
StakingBalance { date: '2024-08-14T00:00:00.000Z' address: '0x80000001677f23a227dfed6f61b132d114be83b8ad0aa5f3c5d1d77e6ee0bf5f73b0af750cc34e8f2dae73c21dc36f4a' bondedStake: '32 ETH' unbondedBalance: '5.566622738 ETH' participantType: 'VALIDATOR' }
StakingBalance { date: '2024-08-13T00:00:00.000Z' address: '0x80000001677f23a227dfed6f61b132d114be83b8ad0aa5f3c5d1d77e6ee0bf5f73b0af750cc34e8f2dae73c21dc36f4a' bondedStake: '32 ETH' unbondedBalance: '5.564549466 ETH' participantType: 'VALIDATOR' }
StakingBalance { date: '2024-08-12T00:00:00.000Z' address: '0x80000001677f23a227dfed6f61b132d114be83b8ad0aa5f3c5d1d77e6ee0bf5f73b0af750cc34e8f2dae73c21dc36f4a' bondedStake: '32 ETH' unbondedBalance: '5.562466119 ETH' participantType: 'VALIDATOR' }
StakingBalance { date: '2024-08-11T00:00:00.000Z' address: '0x80000001677f23a227dfed6f61b132d114be83b8ad0aa5f3c5d1d77e6ee0bf5f73b0af750cc34e8f2dae73c21dc36f4a' bondedStake: '32 ETH' unbondedBalance: '5.560385432 ETH' participantType: 'VALIDATOR' }
StakingBalance { date: '2024-08-10T00:00:00.000Z' address: '0x80000001677f23a227dfed6f61b132d114be83b8ad0aa5f3c5d1d77e6ee0bf5f73b0af750cc34e8f2dae73c21dc36f4a' bondedStake: '32 ETH' unbondedBalance: '5.558446722 ETH' participantType: 'VALIDATOR' }
StakingBalance { date: '2024-08-09T00:00:00.000Z' address: '0x80000001677f23a227dfed6f61b132d114be83b8ad0aa5f3c5d1d77e6ee0bf5f73b0af750cc34e8f2dae73c21dc36f4a' bondedStake: '32 ETH' unbondedBalance: '5.556376299 ETH' participantType: 'VALIDATOR' }
```

#### Static Method
```
async function main() {
    let now = new Date();
    let tenDaysAgo = new Date();
    tenDaysAgo.setDate(now.getDate() - 10);
    const coinbase = Coinbase.configureFromJson({ filePath: "", basePath: "" });
    
    let stakingBalances = await StakingBalance.list(
        "ethereum-mainnet", "eth",
        "0x80000001677f23a227dfed6f61b132d114be83b8ad0aa5f3c5d1d77e6ee0bf5f73b0af750cc34e8f2dae73c21dc36f4a",
        tenDaysAgo.toISOString(), now.toISOString(),
    );
    
    // Loop through the historical staking balances and print each balance
    stakingBalances.forEach(stakingBalance => console.log(stakingBalance));
}


main();
```

External Address Class
```
const coinbase = Coinbase.configureFromJson({ filePath: "", basePath: "" });

const addr = new ExternalAddress(Coinbase.networks.EthereumMainnet, "0x80000001677f23a227dfed6f61b132d114be83b8ad0aa5f3c5d1d77e6ee0bf5f73b0af750cc34e8f2dae73c21dc36f4a") 
addr.historicalStakingBalances(Coinbase.assets.Eth, "2024-07-01T00:00:00Z", "2024-07-05T00:00:00Z").then(balances => { balances.forEach(balance => {console.log(balance)}); } ).catch(error => { console.log(error); });
```

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

**Review by drohit-cb — APPROVED** [review]

**Review by John-peterson-coinbase — APPROVED** [review]

File: src/coinbase/staking_balance.ts [hunk 1]
```diff
@@ -41,9 +41,9 @@ export class StakingBalance {
       const page = queue.shift();
 
       const response = await Coinbase.apiClients.stake!.fetchHistoricalStakingBalances(
-        address,
         networkId,
         assetId,
+        address,
```
Comment (John-peterson-coinbase) [inline src/coinbase/staking_balance.ts:46]: Can we change these to be `addressId`?

File: src/coinbase/balance.ts [hunk 2]
```diff
@@ -63,13 +63,4 @@ export class Balance {
     const asset = Asset.fromModel(model.asset);
     return new Balance(new Decimal(model.amount), asset.getAssetId(), asset);
   }
-
-  /**
-   * Print the Balance as a string.
-   *
-   * @returns The string representation of the Balance.
-   */
-  public toString(): string {
```
[the remark this answers is not in the document]
Comment (marchsu) [reply src/coinbase/balance.ts:72]: this method is added by me only for `stakingBalance`, I am not sure how the other balance would like to define this `toString()` method, or do you think we should move those in `StakingBalance` to here and make it a default behavior?

[the remark this answers is not in the document]
Comment (drohit-cb) [reply src/coinbase/balance.ts:72]: Got it. Happy to keep it limited to staking balances for now.

File: src/coinbase/staking_balance.ts [hunk 3]
```diff
@@ -22,15 +22,15 @@ export class StakingBalance {
...
    * @param endTime - The end time.
    * @returns The staking balances.
    */
   public static async list(
     networkId: string,
     assetId: string,
-    address: string,
+    addressId: string,
     startTime: string,
     endTime: string,
```
Comment (alex-stone) [inline src/coinbase/staking_balance.ts:35]: Can we make this take an object v.s. parameters?

6 parameters is quite a big to have contextual ordering for

**Review by alex-stone — APPROVED** [review]
Change looks good, as we go to `0.1.0` would be nice to update the initial request to take an object to be more usable .

**Pull request merged**